### ISSUE TITLE
systemctl disable on statically enabled units should log a useful message

### ISSUE THREAD

**Issue by jsynacek** [systemd/systemd#9074]
**Description**
Running `systemctl disable sys-kernel-debug.mount` does nothing. Expected behaviour is to remove all relevant symlinks as is expected from the disable operation.

Reproducible with the current master (commit da6c7806fd6b984e98621a844305b2128b897a39).

**Steps to reproduce**
1) Compile and install systemd.
2) Check that the unit file is there.
```
# ls -la /usr/lib/systemd/system/sysinit.target.wants/sys-kernel-debug.mount
lrwxrwxrwx 1 root root 25 May 23 12:52 /usr/lib/systemd/system/sysinit.target.wants/sys-kernel-debug.mount -> ../sys-kernel-debug.mount
```
3) Disable the mount unit.
```
# systemctl disable sys-kernel-debug.mount
```
4) Nothing really happened. The symlink is still there.
```
# ls -la /usr/lib/systemd/system/sysinit.target.wants/sys-kernel-debug.mount
lrwxrwxrwx 1 root root 25 May 23 12:52 /usr/lib/systemd/system/sysinit.target.wants/sys-kernel-debug.mount -> ../sys-kernel-debug.mount
```
Maybe that's because the `sys-kernel-debug.mount` doesn't have the Install section. Let's try adding that to the unit file.

5) Edit `/usr/lib/systemd/system/sysinit.target.wants/sys-kernel-debug.mount` and add the following content to the end of the file:
```
[Install]
WantedBy=sysinit.target
```
6) Restart init.
`# init q`

7) Disable the mount unit.
```
# systemctl disable sys-kernel-debug.mount
```
8) Nothing happened, again. The symlink is still there.
```
# ls -la /usr/lib/systemd/system/sysinit.target.wants/sys-kernel-debug.mount
lrwxrwxrwx 1 root root 25 May 23 12:52 /usr/lib/systemd/system/sysinit.target.wants/sys-kernel-debug.mount -> ../sys-kernel-debug.mount
```

After step 8), I expect that the symlink `/usr/lib/systemd/system/sysinit.target.wants/sys-kernel-debug.mount` should not exist, that is, it was removed by the disable operation.

**Comment by mbiebl** [comment]
sys-kernel-debug.mount is statically enabled. As you noticed the symlinks are in /usr, not /etc.
So an [Install] section will not help, as those symlinks are kept in /etc.
You could try to mask the mount unit though.

**Comment by poettering** [comment]
yes, it's a statically enabled unit. If you really want to get rid of it, use "systemctl mask".

I figure we should output something more useful when one tries to disable a static unit though

**Comment by jsynacek** [comment]
Also, you might consider changing the docs a bit, as it is not perfectly clear (to me, at least) why it's not working. The docs say the following about the `disable` operation (`man systemctl`):

> This removes all symlinks to the unit files backing the specified units from the unit configuration directory, and hence undoes any changes made by enable or link.

Clearly, not all symlinks are removed and I count `/usr/lib/systemd/system` as a configuration directory.

**Comment by poettering** [comment]
So, actually, this is a duplicate of #4012. Closing hence.
